### PR TITLE
33916: CE eliminated calls deprecated methods of the phpunit/phpunit

### PR DESCRIPTION
**Pull request by anzin**
### Description (*)
I've eliminate calls deprecated methods of the phpunit/phpunit, now it provokes warning, but it will provoke error in future.

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/599
https://github.com/magento/partners-magento2b2b/pull/585
https://github.com/magento/adobe-stock-integration/pull/1864

### Fixed Issues (if relevant)

1. Fixes magento/magento2#33916

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php [hunk 1]
```diff
@@ -177,37 +185,43 @@ public function testValidate($attributeData, $data, $expected)
...
-    public function testValidateUnique()
+    /**
+     * @return void
+     */
+    public function testValidateUnique(): void
     {
         $this->_attribute->expects($this->any())->method('getName')->willReturn('attribute_name');
-        $this->_attribute->expects($this->at(1))->method('getIsRequired');
-        $this->_attribute->expects($this->at(2))->method('getIsUnique')->willReturn(true);
+        $this->_attribute
```
Comment (BarnyShergold) [inline app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php]: Single method call so should be on one line

File: app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php [hunk 2]
```diff
@@ -219,16 +233,24 @@ public function testValidateUnique()
...
+    /**
+     * @return void
+     */
+    public function testValidateUniqueException(): void
     {
         $this->expectException('Magento\Framework\Exception\LocalizedException');
         $this->_attribute->expects($this->any())->method('getName')->willReturn('attribute_name');
-        $this->_attribute->expects($this->at(1))->method('getIsRequired');
-        $this->_attribute->expects($this->at(2))->method('getIsUnique')->willReturn(true);
+        $this->_attribute
```
Comment (BarnyShergold) [inline app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php]: Single method call so should be on one line

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
A superb PR - not just removing all the deprecated methods but refactoring and cleaning up the DOCBLOCKS & return types on prototypes.

I only found 2 things suggested to change.

But - as you have done such a great job cleaning up the classes and adding things that are now best practice, can I ask - should not all "protected" methods be changed to "private"?

Same thing goes for class variables?
EG 
protected $objectManagerHelper;

to 

private $objectManagerHelper;

**Comment by anzin** [comment]
@magento run WebAPI Tests, Functional Tests EE

**Comment by anzin** [comment]
@magento run  Functional Tests EE

File: app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php (hunk 2 above)
Comment (anzin) [reply app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php]: Hello @BarnyShergold, thanks for your review, I corrected the code according to the recommendations, please check if everything well.

Thank you.

cc @andrewbess

**Comment by anzin** [comment]
@magento run all tests

**Comment by anzin** [comment]
@magento run Functional Tests CE

File: app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php (hunk 1 above)
Comment (BarnyShergold) [reply app/code/Magento/Catalog/Test/Unit/Model/Category/Attribute/Backend/SortbyTest.php]: This one doesn't seem to show up as fixed?

**Comment by anzin** [comment]
@magento run all tests

**Comment by anzin** [comment]
@magento run Functional Tests CE

**Review by andrewbess — APPROVED** [review]

**Review by eduard13 — APPROVED** [review]

**Pull request merged**